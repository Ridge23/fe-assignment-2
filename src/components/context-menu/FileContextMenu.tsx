import React from 'react';
import { MenuItem } from "react-contextmenu";

import DeleteIcon from 'components/icons/DeleteIcon';
import RenameIcon from 'components/icons/RenameIcon';
import ContextMenu from './ContextMenu';
interface IFileContextMenu {
    id: number;
    onRenameClick: () => void;
    onDeleteClick: () => void;
}

export default function FileContextMenu({ id, onRenameClick, onDeleteClick }: IFileContextMenu): JSX.Element {
    return (
        <ContextMenu id={`file_menu_${id}`} >
            <MenuItem onClick={onRenameClick}>
                <RenameIcon /> Rename
            </MenuItem>
            <MenuItem onClick={onDeleteClick}>
                <DeleteIcon /> Delete
            </MenuItem>
        </ContextMenu>);
}