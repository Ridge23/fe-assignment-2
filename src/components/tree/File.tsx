import React from 'react';

import './sass/file.scss';

export default function File({ name }): JSX.Element {
    return <li key={`file-${name}`} className="file">{name}</li>;
}