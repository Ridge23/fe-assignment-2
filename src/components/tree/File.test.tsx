import React from 'react';
import { shallow } from 'enzyme';
import { ContextMenuTrigger } from "react-contextmenu";

import FileContextMenu from 'components/context-menu/FileContextMenu';
import File from './File';

const noop = () => {};

describe('File', () => {
    it('should render correctly', () => {
        const component = shallow(<File id={10} name='test.js' onDelete={noop} onRename={noop} />);
        const title = component.find('.file__title');

        expect(component.find('li')).toHaveLength(1);
        expect(component.find(ContextMenuTrigger)).toHaveLength(1);
        expect(component.find(FileContextMenu)).toHaveLength(1);
        expect(component.find('li')).toHaveProperty('key');
        expect(title.text()).toEqual('test.js');
        expect(component).toMatchSnapshot();
    });
})