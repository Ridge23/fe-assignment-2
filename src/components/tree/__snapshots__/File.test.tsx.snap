// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File should render correctly 1`] = `
<li
  className="file"
  key="file-test.js"
>
  <ContextMenuTrigger
    attributes={Object {}}
    collect={[Function]}
    disable={false}
    disableIfShiftIsPressed={false}
    holdToDisplay={1000}
    id="file_menu_10"
    mouseButton={2}
    posX={0}
    posY={0}
    renderTag="div"
  >
    <div
      className="file__title"
    >
      test.js
    </div>
  </ContextMenuTrigger>
  <FileContextMenu
    id={10}
    onDeleteClick={[Function]}
    onRenameClick={[Function]}
  />
</li>
`;
